File: Weeks/Week01/src/6_InverseTranspose.py
import numpy as np

np.set_printoptions(formatter={'float': lambda x: "{0:0.6f}".format(x)})

# % Initialize matrix A 
A = np.matrix([[1,2,0], [0,5,6], [7,0,9]])
print(A, '\n')

# % Transpose A 
A_trans = A.T
print(A_trans, '\n')

# % Take the inverse of A 
A_inv = A.I
print(A_inv, '\n')

# % What is A^(-1)*A? 
A_invA = np.dot(A_inv, A)
print(A_invA, '\n')

File: Weeks/Week01/src/6_InverseTranspose.cpp
#include <armadillo>
#include <iostream>

using namespace arma;
using namespace std;

// ./cpp_shell.sh 6_InverseTranspose

int main() {
  mat A = {
    {1, 2, 0},
    {0, 5, 6},
    {7, 0, 9}
  };
  cout << "Matrix A =" << endl;
  A.print();

  mat A_trans = trans(A);
  cout << "Transpose of A =" << endl;
  A_trans.print();

  mat A_inv = inv(A);
  cout << "Inverse of A =" << endl;
  A_inv.print();

  mat A_invA = inv(A) * A;
  cout << "What is A^(-1) * A =" << endl;
  A_invA.print();

  return 0;
}